plugins/woocommerce/includes/admin/class-wc-admin-assets.php

```diff
@@ -56,7 +56,10 @@ public function admin_styles() {
 
 			if ( $screen && $screen->is_block_editor() ) {
 				wp_register_style( 'woocommerce-general', WC()->plugin_url() . '/assets/css/woocommerce.css', array(), $version );
+				wp_register_style( 'woocommerce-blocktheme', WC()->plugin_url() . '/assets/css/woocommerce-blocktheme.css', array(), $version );
 				wp_style_add_data( 'woocommerce-general', 'rtl', 'replace' );
+				wp_style_add_data( 'woocommerce-blocktheme', 'rtl', 'replace' );
+				wp_enqueue_style( 'woocommerce-blocktheme' );
```

Comment:
I think you should wrap the lines you added in a `if ( wc_current_theme_is_fse_theme() )` check, so they only run if the site has a block theme enabled.